src/ui/public/directives/paginated_selectable_list.js

```diff
@@ -0,0 +1,56 @@
+import _ from 'lodash';
+import uiModules from 'ui/modules';
+import paginatedSelectableListTemplate from 'ui/partials/paginated_selectable_list.html';
+
+const module = uiModules.get('kibana');
+
+module.directive('paginatedSelectableList', function (kbnUrl) {
+
+  return {
+    restrict: 'E',
+    scope: {
+      perPage: '=',
+      list: '=',
+      userMakeUrl: '=',
+      userOnSelect: '='
+    },
+    template: paginatedSelectableListTemplate,
+    controller: function ($scope, $element, $filter) {
+      $scope.perPage = $scope.perPage || 10;
+
+      $scope.hits = $scope.list || [];
+
+      $scope.hitCount = $scope.hits.length;
+
+      $scope.makeUrl = function (hit) {
+        if ($scope.userMakeUrl) {
+          return $scope.userMakeUrl(hit);
+        }
+        return '#';
+      };
+
+      $scope.onSelect = function (hit, $event) {
+        if ($scope.userOnSelect) {
+          return $scope.userOnSelect(hit, $event);
+        }
+
```

Comment:
You can get rid of everything below this if you move to the system described above, eg divorcing makeUrl and onSelect.